app/src/main/java/com/keylesspalace/tusky/ViewMediaActivity.kt

```diff
@@ -0,0 +1,249 @@
+import android.support.v4.content.FileProvider
+import android.support.v4.view.ViewPager
+import android.support.v7.widget.Toolbar
+import android.util.Log
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import com.keylesspalace.tusky.BuildConfig.APPLICATION_ID
+
+import com.keylesspalace.tusky.fragment.ViewMediaFragment
+import com.keylesspalace.tusky.pager.AvatarImagePagerAdapter
+import com.keylesspalace.tusky.pager.ImagePagerAdapter
+import com.keylesspalace.tusky.util.CollectionUtil.map
+import com.keylesspalace.tusky.util.MediaUtils
+import com.keylesspalace.tusky.view.ImageViewPager
+import com.keylesspalace.tusky.viewdata.AttachmentViewData
+import com.squareup.picasso.Picasso
+import com.squareup.picasso.Target
+
+import java.io.File
+import java.io.FileNotFoundException
+import java.io.FileOutputStream
+import java.io.IOException
+import java.util.ArrayList
+
+class ViewMediaActivity : BaseActivity(), ViewMediaFragment.PhotoActionsListener {
+    companion object {
+        private const val EXTRA_ATTACHMENTS = "attachments"
+        private const val EXTRA_ATTACHMENT_INDEX = "index"
+        private const val EXTRA_AVATAR_URL = "avatar"
+        private const val TAG = "ViewMediaActivity"
+
+        @JvmStatic
+        fun newIntent(context: Context?, attachments: List<AttachmentViewData>, index: Int): Intent {
+            val intent = Intent(context, ViewMediaActivity::class.java)
+            intent.putParcelableArrayListExtra(EXTRA_ATTACHMENTS, ArrayList(attachments))
+            intent.putExtra(EXTRA_ATTACHMENT_INDEX, index)
+            return intent
+        }
+
+        fun newAvatarIntent(context: Context, url: String): Intent {
+            val intent = Intent(context, ViewMediaActivity::class.java)
+            intent.putExtra(EXTRA_AVATAR_URL, url)
+            return intent
+        }
+    }
+
+    private lateinit var viewPager: ImageViewPager
+    private lateinit var toolbar: Toolbar
+
+    private var attachments: ArrayList<AttachmentViewData>? = null
+
+    private var toolbarVisible = true
+    private val toolbarVisibilityListeners = ArrayList<ToolbarVisibilityListener>()
+
+    interface ToolbarVisibilityListener {
+        fun onToolbarVisiblityChanged(isVisible: Boolean)
+    }
+
+    fun addToolbarVisibilityListener(listener: ToolbarVisibilityListener): Function0<Boolean> {
+        this.toolbarVisibilityListeners.add(listener)
+        listener.onToolbarVisiblityChanged(toolbarVisible)
+        return { toolbarVisibilityListeners.remove(listener) }
+    }
+
+    fun isToolbarVisible(): Boolean {
+        return toolbarVisible
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_view_media)
+
+        supportPostponeEnterTransition()
+
+        // Obtain the views.
+        toolbar = findViewById(R.id.toolbar)
+        viewPager = findViewById(R.id.view_pager)
```

Comment:
I'd prefer if you used the [Android extensions](https://kotlinlang.org/docs/tutorials/android-plugin.html#using-kotlin-android-extensions) for this (import the layout and then you can reference the views by their id in Kotlin, no `findViewById` necessary). Rename the ids in the layout file so the look nice in Kotlin.